airbyte-integrations/connectors/source-mixpanel/source_mixpanel/manifest.yaml

```diff
@@ -210,6 +219,17 @@ definitions:
         http_method: POST
       paginator:
         $ref: "#/definitions/paginator"
+      record_selector:
+        $ref: "#/definitions/selector"
+        record_filter:
+          condition: |
+            {%- set partition_state = namespace(value=config.start_date) -%}
+            {%- for state in stream_state.get('states', []) -%}
```

Comment:
@midavadim this is a pretty complicated condition. can you add tests to cover all branches?

Reply:
I have added unittests for this case